src/webui/webapplication.cpp

```diff
@@ -774,6 +777,33 @@ void WebApplication::action_command_bottomPrio()
     BitTorrent::Session::instance()->bottomTorrentsPriority(hashes);
 }
 
+void WebApplication::action_command_setLocation()
+{
+    CHECK_URI(0);
+    CHECK_PARAMETERS("hashes" << "location");
+
+    QStringList hashes = request().posts["hashes"].split("|");
+    QString newLocationPath = request().posts["location"].trimmed();
+
+    // check location exists
+    QFileInfo newLocation(newLocationPath);
+    if (!newLocation.exists()) {
```

Comment:
Take a look at `TransferListWidget::setSelectedTorrentsLocation()`.
`QDir` is more suitable IMO.